gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/spi/PubSubRpc.java

```diff
@@ -69,4 +70,6 @@
   Future<PullResponse> pull(PullRequest request);
 
   Future<Empty> modify(ModifyPushConfigRequest request);
+
+  ScheduledExecutorService executor();
```

Comment:
Yes I believe we should allow users to set the executor via the service options. That way the same executor can be reused across services.